lib/puppet/file_system/windows.rb

```diff
@@ -3,6 +3,20 @@
 
 class Puppet::FileSystem::Windows < Puppet::FileSystem::Posix
 
+  def expanded_path_string(path)
+    string_path = path.to_s
+    # if no tildes, nothing to expand, return original string
+    # WARN: this will return a / on Windows if passed a /
+    return string_path if string_path !~ /~/
+
+    expanded = ''
+    Pathname.new(string_path).descend do |f|
+      expanded = File.expand_path(File.join(expanded, File.basename(f)))
```

Comment:
I don't believe it will (there are other similar assumptions that the code already makes around these sorts of questions)